### ISSUE TITLE
Free trial killed my purchase.

### ISSUE THREAD

**Issue by Grunerd** [SQLiteFlow/SQLiteFlow-Issues#25]
The Restore Previous Purchase-Button gives me a spinning loading animation after entering my credentials and then nothing happens. When I close the window once that animation is finished, it closes the app. I have no interest in "trying" an app that I purchased.

And in addition to that: I effing hate apps that go with this approach. Because it always comes with issues. Offering a free version and a separate full version would've been the better choice. Swapped my 5-star-rating for a 1-star rating with a warning until this issue is resolved.

<img width="565" alt="Screenshot 2019-04-24 at 08 18 47" src="https://user-images.githubusercontent.com/11262143/56636691-e3be2780-6669-11e9-82fe-5c6c9626796a.png">
<img width="565" alt="Screenshot 2019-04-24 at 08 18 49" src="https://user-images.githubusercontent.com/11262143/56636814-3861a280-666a-11e9-8596-f77dd8851c11.png">

**Comment by SQLiteFlow** [comment]
@Grunerd 
Don't worry, this is a knowing issue. A fix version 3.5.1 has been submit to Apple's App Review team.
After you update to this version, everything will be ok.
Now you can just click 'Get the Free Trial' button to using the app.

**Comment by SQLiteFlow** [comment]
@Grunerd 
And for respond the addition one, I am considering your advice, my thought of the Trial approach is that it may let users have evaluation about the software before they know they want to purchase it, I was thought it's good, I may be wrong, and what the difference between this approach and offering a free version and a separate full version approach?

**Comment by Grunerd** [comment]
Thank you very much for the quick reply. I hope the patch will be out very soon. It was quite a shock when I updated and simply "couldn't access what I paid for". 

It is very much a psychological issue for customers these days and I'm no exception to that. I have been burned quite a lot, lately with app developers switching their pricing / monetization schemes. 1Password for example makes it almost "impossible" to buy a standalone license, forcing their cloud service and subscription model down your throat. It still exists so that they can say they offer it when criticised - but you have to either dive knee-deep into their website or ask the support in a long and exhausting conversation till they finally let you get "just the full app" and nothing else. 1Password used to be a full standalone application on the app store. Many people updated from 1Password 6 to 1Password 7 just to find out that the most basic functionality - Vaults (local libraries for all your passwords) - were gone. And that switch basically broke their 1Password 6 installation too, forcing them to remove everything and set it up again when they decided to roll back. It's like that one feature that made 1Password "1Password". 1Password 7 on the App Store now _only_ works with the Cloud. 

On the Mac App Store, when you buy a full app you should get just that - a full app. But the moment Microtransactions and / or  In-App-Purchases are introduced, that app feels torn apart and the customer doesn't know what the future will bring. Will the developer turn to a subscription based model? Will you only get basic updates and new, important or cool features will be hidden behind another paywall? Will features that have been part of the app from the beginning suddenly be put behind a paywall? We've had those things already happen a lot. 

And in your specific case, it felt like a "betrayal" when you out of nowhere went from a commonly priced full application to something that suddenly used In-App-Purchases. Especially when this step apparently "broke" the app with that step. That's two no-gos in one.

Another issue is that with the scheme you set up - Free and Paid App combined into one thing, it's easier to _fake_ good reviews, because no one needs to actually buy the app to review it. And customers know that. When there are only paid users reviewing, you get less reviews, but it feels more genuine, as those customers "should" usually be customers who actually paid for that product and therefore committed to the product.  

I enjoy SQLiteFlow and I will use it in the future. In my initial review, I wrote that your app is worth way more than what I paid for it and I think the new price is pretty good.

**Comment by rysmax** [comment]
I bought SQL Flow before (3$) but after update to new version I get the dialog box with 2 button: "Restore Previous Purchase" and "Get The Free Trial".
I tried click to "Restore Previous Purchase" but this dialog do nothing.
And "Get The Free Trial" do nothing too.
It's looks like hoax to take money again.
My expectations were that when I press the button "Restore Previous Purchase" I get working application without additional payment FOR USERS WHICH BOUGHT THIS APPLICATION IN THE PAST.

**Comment by SQLiteFlow** [comment]
Hello @rysmax 
‘Get the Free Trial' should bring you to the app, what the app responds when you click the 'Get the Free Trial' button?

> In version 3.5.1, I fix an issue that may cause paid users see getting trial or getting full unlock screen.

For users who experiencing this issue, you can just click ‘Get the Free Trial’ to continue using the app as a workaround before updating to the fixed version.

**Comment by Grunerd** [comment]
@rysmax As the developer mentioned above, it's a known issue and will be fixed with the next update that's already on the way. At least for now, it doesn't seem that we have to pay again. 

@SQLiteFlow Another issue regarding "trust" is, that "Get the Free Trial" also asks for your password, because it technically is a free in-app-purchase. So it doesn't take you to the app right away, but asks for app store authorisation.

My workaround for now is that I rolled back to 3.3.1 via my TimeMachine backup and I'll stay at that version until this is sorted out.

**Comment by rysmax** [comment]
> Hello @rysmax
> ‘Get the Free Trial' should bring you to the app, what the app responds when you click the 'Get the Free Trial' button?
> 
> > In version 3.5.1, I fix an issue that may cause paid users see getting trial or getting full unlock screen.
> > Version 3.5.1 is pending release.
> 
> > For users who experiencing this issue, you can just click ‘Get the Free Trial’ to continue using the app as a workaround before updating to the fixed version.

Unfortunately nothing going on - button "Get Free Trial" not working.
Throbber spin around infinitely.

**Comment by rysmax** [comment]
> My workaround for now is that I rolled back to 3.3.1 via my TimeMachine backup and I'll stay at that version until this is sorted out.

I think the same

**Comment by SQLiteFlow** [comment]
@Grunerd 
Thank you very much too for the reply. It was quite a shock to me too when I updated SQLiteFlow from the production Mac App Store server today, since everything was good in the testing Mac App Store server, it's my fault that I made the mistake. I hope the fixed version will be available soon too.

I think I get your point. I will let future SQLiteFlow pricing scheme loyal to exists(or already paid) users, since they paid for a full standalone version, they will always have a full standalone version.

I may understand developers turn their softwares to a subscription based model for keeping their business going, SQLiteFlow currently not have plan to change the pricing schemes to subscription model, even if it does, I will not let it affect to exists users.

**Comment by rysmax** [comment]
> @Grunerd
> Thank you very much too for the reply. It was quite a shock to me too when I updated SQLiteFlow from the production Mac App Store server today, since everything was good in the testing Mac App Store server, it's my fault that I made the mistake. I hope the fixed version will be available soon too.
> 
> I think I get your point. I will let future SQLiteFlow pricing scheme loyal to exists(or already paid) users, since they paid for a full standalone version, they will always have a full standalone version.
> 
> I may understand developers turn their softwares to a subscription based model for keeping their business going, SQLiteFlow currently not have plan to change the pricing schemes to subscription model, even if it does, I will not let it affect to exists users.

Thank you for respecting users and quick response!!!
5 stars to you in AppStore and my recommendations for using this application like best SQLite editor on Mac in my new CMS/CMF Effcore (coming soon) and other projects.

**Comment by SQLiteFlow** [comment]
@Grunerd 
Yeah, but this one is kind of out of control, since this is the App Store way to get a 'Free Trial' receipt. If one day App Store don't require password for free In-App purchases, SQLiteFlow may benefit from it.

> Another issue regarding "trust" is, that "Get the Free Trial" also asks for your password, because it technically is a free in-app-purchase.

**Comment by SQLiteFlow** [comment]
@rysmax 
Thank you for the reply, and the rating, even I don't think I deserve it.

**Comment by Grunerd** [comment]
3.5.1 hit the App Store. I updated and was thrown into the main application immediately. So far, it looks like everything works. But if you hear from me in 2 weeks angrily waving my hands around, my full purchase wasn't recognised and I'm stuck in demo mode. :P I trust you that this scenario won't happen. 

So long, thank you very much for the conversation and of course for the relatively quick fix. Keep up the good work! I just updated my review up to 5 stars again - should be visible soon.

**Comment by SQLiteFlow** [comment]
@Grunerd 
Thank you for your kind responses, and I will try my best to make SQLiteFlow better and better.

In addition, the latest version is 3.5.2 now, which also fixed an issue that may cause Data Viewer show empty records, if you also experienced this issue, it's recommend to update to this version.

**Comment by SQLiteFlow** [comment]
@rysmax 
Issue fixed, the latest version 3.5.2 is available on App Store now.
